iceoryx2-services/discovery/tests/service_discovery_service_tests.rs

```diff
@@ -196,4 +197,67 @@ mod service_discovery_service {
             test_fail!("expected DiscoveryEvent::Added for the internal service")
         }
     }
+
+    #[test]
+    fn get_current_discovery_states() -> Result<(), Box<dyn core::error::Error>> {
+        const NUM_SERVICES: usize = 100;
+
+        let iceoryx_config = generate_isolated_config();
+
+        // create some services in the main thread
+        let node = NodeBuilder::new()
+            .config(&iceoryx_config)
+            .create::<ipc::Service>()
+            .unwrap();
+
+        let service_names: Vec<_> = (0..NUM_SERVICES).map(|_| generate_name()).collect();
+
+        let _created_services: Vec<_> = service_names
+            .iter()
+            .map(|name| {
+                node.service_builder(name)
+                    .publish_subscribe::<u64>()
+                    .create()
+                    .unwrap()
+            })
+            .collect();
+
+        let discovery_config = Config {
```

Comment:
If you add the config `initial_max_response_size` you could set it to `NUM_SERVICES` so allocation only occurs once (when the server is created).